src/persistentReducedStream.ts

```diff
@@ -14,26 +13,68 @@ import { defaultErrorSubject } from './internal/defaultErrorSubject';
 import { markName } from './internal/markers';
 import { tag } from 'rxjs-spy/operators';
 
-/**
- * A hot, persistent observable with a state field
- */
-export class StateStream<State> extends Observable<State>
-  implements SubscriptionLike {
+export class PersistentReducedStateStream<State> extends Observable<State> {
   private subject: BehaviorSubject<State>;
-  private subscription: Subscription;
-
-  constructor(initialState: State, reducedState$: Observable<State>) {
-    super();
-    this.subject = new BehaviorSubject(initialState);
+  private reducerSubscription?: Subscription;
+  public name: string;
+  private reducers: RegisteredReducer<State, any>[];
+  private errorSubject: Subject<any>;
 
+  /**
+   * Subscribes to the action stream and starts reducing this state stream.
+   * startReducing does not always immediately emit a value since the value
+   * might have already been emitted when the state stream was created.
+   *
+   * initialState can be passed to reset the state of the stream to some other value.
+   * This can be useful for hot-reloading
+   *
+   * startReducing will recreate the inner subject if it was previously closed
+   */
+  startReducing = (action$: ActionStream, initialState?: State) => {
```

Comment:
`startReducing` also supports "restarting" of reducing if the inner subject is closed and `initialState` is passed, which is nice for hot-reloading: https://github.com/ardoq/ardoq-front/pull/7135/files#diff-ff0d9d26e235633b69491de7bc7b6cf5L27

Alternatively, we could split this into two methods `startReducing` and `restartReducing` to make it more explicit. Thoughts?